### PR TITLE
use pathlib

### LINKED ISSUE #240: CLN use Path.write_text where possible

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #240

### CODE REVIEW

File: nbqa/replace_source.py [hunk 1]
```diff
@@ -88,8 +88,7 @@ def main(python_file: "Path", notebook: "Path", notebook_info: NotebookInfo) ->
     with open(notebook) as handle:
         notebook_json = json.load(handle)
```
Comment (girip11) [inline nbqa/replace_source.py:89]: You could also write this using `json.loads(notebook.read_text())`

File: tests/test_configs_work.py [hunk 2]
```diff
@@ -23,8 +23,7 @@ def test_configs_work(capsys: "CaptureFixture") -> None:
     capsys
         Pytest fixture to capture stdout and stderr.
     """
-    with open(".flake8", "w") as handle:
-        handle.write("[flake8]\nignore=F401\nselect=E303\nquiet=1\n")
+    Path(".flake8").write_text("[flake8]\nignore=F401\nselect=E303\nquiet=1\n")
```
Comment (girip11) [inline tests/test_configs_work.py:26]: How about using dedent for this too?

**Review by girip11 — CHANGES_REQUESTED** [review]

File: nbqa/replace_source.py (hunk 1 above)
Comment (MarcoGorelli) [reply nbqa/replace_source.py:89]: nice!

Comment (MarcoGorelli) [reply nbqa/replace_source.py:89]: seems there's a _slight_ performance hit from this, but it's probably fine for the sake of readability
```python
In [8]: %timeit json.loads(Path(notebook).read_text())                                                                                                                                                    
41 µs ± 159 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [9]: %timeit with open(notebook) as handle: json.load(handle)                                                                                                                                          
31.7 µs ± 129 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

**Review by girip11 — APPROVED** [review]

**Pull request merged**